aeron-cluster/src/main/java/io/aeron/cluster/ConsensusModuleAgent.java

```diff
@@ -1339,7 +1342,9 @@ long prepareForNewLeadership(final long logPosition, final long nowNs)
             clearSessionsAfter(logPosition);
             for (int i = 0, size = sessions.size(); i < size; i++)
             {
-                sessions.get(i).disconnect(aeron, ctx.countedErrorHandler());
+                final ClusterSession session = sessions.get(i);
+                session.unlinkIngressImage();
```

Comment:
Should `ClusterSession::unlinkIngressImage` be part of `ClusterSession::disconnect`?  Is there a reason that it needs to be separate?  Also, should we be unlinking in `ConsensusModuleAgent::onSessionClose`?

Reply:
It feels to me that it would be cleaner if it was unlinked on closed as well.  However, I think we should look at that as a separate change.  Not needed for this PR.